src/Model.js

```diff
@@ -436,9 +436,8 @@ class Model {
 
     const attributeNames = options.attributes || this.attributeNames
     const relationshipNames = options.relationships || this.relationshipNames
-
-    const validAttributes = validateProperties(this, attributeNames, attributeDefinitions)
-    const validRelationships = validateProperties(this, relationshipNames, relationshipDefinitions)
+    const validAttributes = attributeDefinitions ? validateProperties(this, attributeNames, attributeDefinitions) : []
+    const validRelationships = relationshipDefinitions ? validateProperties(this, relationshipNames, relationshipDefinitions) : []
```

Comment:
same thing here, what if we made this change instead:

```diff
  get attributeDefinitions () {
    const { type } = this.constructor
-    return schema.structure[type]
+    return schema.structure[type] || {}
  }
```

Reply:
ok

Reply:
Or maybe we should make sure that `schema.structure[type] = {}`.